runtime/router.go

```diff
@@ -99,9 +100,11 @@ func (endpoint *RouterEndpoint) HandleRequest(
 
 // HTTPRouter data structure to handle and register endpoints
 type HTTPRouter struct {
+	gateway                  *Gateway
```

Comment:
Nitpick: you could just save the `logger` here, since only the gateway logger is used.